plugin/documents.py

```diff
@@ -383,6 +383,17 @@ def on_post_save_async(self) -> None:
             # The URI scheme has changed. This means we need to re-determine whether any language servers should
             # be attached to the view.
             sublime.set_timeout(self._reset)
+        # Hide empty diagnostics panels on save if the show_diagnostics_panel_on_save option is enabled.
+        if userprefs().show_diagnostics_panel_on_save > 0:
+            self._hide_empty_diagnostics_panels()
+
+    def _hide_empty_diagnostics_panels(self) -> None:
+        change_count = self.view.change_count()
+        for sb in self.session_buffers_async():
+            if sb.diagnostics_version != change_count or not sb.diagnostics:
+                mgr = sb.session.manager()
+                if mgr:
+                    mgr.hide_diagnostics_panel_async()
```

Comment:
We should only hide the panel if none of the diagnostics (which you can retrieve with `self.diagnostics_async()` match the criteria for opening the panel. So the check have to consider diagnostic's severity levels and not just check if there are no diagnostics.

And, correct me if I'm wrong, but I believe that after saving the document the `sb.diagnostics_version != change_count` will pretty much always be true since the change count just changed and there was not enough time to receive new diagnostics. So there should be no reason for checking it.

Reply:
Fixed the logic (hide only if none of the diagnostics is visible), checked against the severity level setting, and used `self.diagnostics_async()` instead of reinventing the wheel.

Reply:
> Also this has the same issue as previous solution as SessionBuffer of one server can trigger panel close even if other session still has diagnostics.

Just saw your edit, sorry, are all session buffers sharing a single panel or do they have one each?

Reply:
Panel is per window.